linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/RequestFramingFilter.scala

```diff
@@ -0,0 +1,46 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Request, Response, Status}
+import com.twitter.finagle.buoyant.linkerd.Headers
+import com.twitter.logging.Logger
+import com.twitter.util.Future
+
+/**
+ * A filter that fails badly-framed requests.
+ */
+class RequestFramingFilter extends SimpleFilter[Request, Response] {
+  private[this] lazy val log = Logger.get("RequestFramingFilter")
+
+  override def apply(
+    request: Request,
+    service: Service[Request, Response]
+  ): Future[Response] =
+    if (request.headerMap.getAll("Content-Length").length > 1) {
```

Comment:
I think this will fail requests with multiple content length headers that all have the same value.  If I'm reading the spec right, I think that should be valid.